src/test/java/teammates/test/cases/datatransfer/AdminEmailAttributesTest.java

```diff
@@ -0,0 +1,254 @@
+import static teammates.common.util.FieldValidator.INVALID_NAME_ERROR_MESSAGE;
+import static teammates.common.util.FieldValidator.REASON_CONTAINS_INVALID_CHAR;
+import static teammates.common.util.FieldValidator.REASON_EMPTY;
+import static teammates.common.util.FieldValidator.REASON_START_WITH_NON_ALPHANUMERIC_CHAR;
+import static teammates.common.util.FieldValidator.REASON_TOO_LONG;
+import static teammates.common.util.FieldValidator.REGEX_NAME;
+import static teammates.common.util.FieldValidator.SIZE_CAPPED_NON_EMPTY_STRING_ERROR_MESSAGE;
+import static teammates.test.driver.StringHelperExtension.generateStringOfLength;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.appengine.api.datastore.Text;
+
+import teammates.common.datatransfer.attributes.AdminEmailAttributes;
+import teammates.common.util.Const;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.StringHelper;
+import teammates.common.util.TimeHelper;
+import teammates.storage.entity.AdminEmail;
+
+/**
+ * SUT: {@link AdminEmailAttributes}.
+ */
+public class AdminEmailAttributesTest extends BaseAttributesTest {
+
+    private FieldValidator fieldValidator = new FieldValidator();
+    private AdminEmailAttributes adminEmailAttributes;
+    private List<String> addressReceiverListString = Arrays.asList("example1@test.com", "example2@test.com");
+    private List<String> groupReceiverListFileKey = Arrays.asList("listfilekey", "listfilekey");
+    private String subject = "subject of email";
+    private Text content = new Text("valid email content");
+    private Date date = new Date();
+
+    @BeforeClass
+    public void classSetup() {
+        adminEmailAttributes = new AdminEmailAttributes(
+                subject, addressReceiverListString, groupReceiverListFileKey, content, date);
+    }
+
+    @Test
+    public void testValidate() throws Exception {
+        ______TS("valid admin email");
+
+        assertTrue("Valid input", adminEmailAttributes.isValid());
+        List<String> errorList = adminEmailAttributes.getInvalidityInfo();
+        assertTrue("Valid input should return an empty list of errors", errorList.isEmpty());
+
+        ______TS("success: subject max length");
+
+        String subjectMaxLengthString = generateStringOfLength(EMAIL_SUBJECT_MAX_LENGTH);
+        AdminEmailAttributes validAttributesSubjectLength = new AdminEmailAttributes(
+                subjectMaxLengthString, addressReceiverListString, groupReceiverListFileKey, content, date);
+
+        assertTrue("Valid input", validAttributesSubjectLength.isValid());
+        List<String> emailErrorList = validAttributesSubjectLength.getInvalidityInfo();
+        assertTrue("Valid input should return an empty list of errors", emailErrorList.isEmpty());
+
+        ______TS("invalid admin email");
```

Comment:
There is no need for this. The `failure:` prefix already tells that it is invalid.